src/powerquery-language-services/analysis/analysis.ts

```diff
@@ -6,7 +6,7 @@ import type { CompletionItem, Hover, SignatureHelp } from "vscode-languageserver
 import { IDisposable } from "../commonTypes";
 
 export interface Analysis extends IDisposable {
-    getCompletionItems(): Promise<ReadonlyArray<CompletionItem>>;
+    getCompletionItems(): Promise<CompletionItem[]>;
```

Comment:
We seem to be inconsistent on the use of `[]` vs. `ReadonlyArray`. I prefer `[]` for the sake of brevity, but I'm not exactly sure what the difference ends up being.

Reply:
In short, use`Readonly<T>` internally, and use `[]` for VSCode APIs.

You've probably noticed, but in general I prefer opt-in mutability. A bit ago I refactored the language-services layer to use `ReadonlyArray<T>` everywhere. I'm pulling back a bit on that so wherever we interface with VSCode APIs it returns `[]` as that's what the API expects.